zebra-state/src/service/finalized_state/disk_format/transparent.rs

```diff
@@ -54,59 +74,114 @@ impl OutputIndex {
         )
     }
 
-    /// Return this index as the native index integer type.
+    /// Return this index as `usize`.
     #[allow(dead_code)]
     pub fn as_usize(&self) -> usize {
         self.0
             .try_into()
             .expect("the maximum valid index fits in usize")
     }
 
-    /// Create a transparent output index from the Zcash consensus integer type.
-    pub fn from_zcash(output_index: u32) -> OutputIndex {
-        OutputIndex(output_index)
+    /// Create a transparent output index from `u64`.
+    #[allow(dead_code)]
+    pub fn from_u64(output_index: u64) -> OutputIndex {
+        OutputIndex(
+            output_index
+                .try_into()
+                .expect("the maximum u64 index fits in the inner type"),
+        )
     }
 
-    /// Return this index as the Zcash consensus integer type.
+    /// Return this index as `u64`.
     #[allow(dead_code)]
-    pub fn as_zcash(&self) -> u32 {
+    pub fn as_u64(&self) -> u64 {
         self.0
+            .try_into()
+            .expect("the maximum valid index fits in u64")
```

Comment:
Suggestion: remove the expect by using `.into()` instead of `.try_into()`.

Reply:
Weird, I'm surprised clippy didn't catch that.

```suggestion
            .into()
```